src/lib/utils/config/src/statics.rs

```diff
@@ -74,6 +75,44 @@ fn create_config() -> ServerConfig {
     }
 }
 
+fn create_whitelist() -> Vec<PlayerIdentity> {
+    let whitelist_location = get_root_path().join("whitelist.json");
+    if !whitelist_location.exists() {
+        if let Err(e) = File::create(&whitelist_location).and_then(|mut file| file.write_all(b"[]"))
+        {
+            error!("Could not create initial white-list file: {e}");
+            return Vec::new();
+        }
+        return Vec::new();
+    }
+
+    let mut file = match File::open(&whitelist_location) {
+        Ok(file) => file,
+        Err(e) => {
+            error!("Could not open white-list file: {e}");
+            return Vec::new();
+        }
+    };
+
+    let mut whitelist_str = String::new();
+    if let Err(e) = file.read_to_string(&mut whitelist_str) {
+        error!("Could not read white-list file: {e}");
+        return Vec::new();
+    }
+
+    if whitelist_str.is_empty() {
+        return Vec::new();
+    }
+    serde_json::from_str::<Vec<PlayerIdentity>>(&whitelist_str).unwrap_or_else(|e| {
```

Comment:
Using json seems kinda redundant. The data is not very complex, a newline/comma/space separated file format would suffice. Not a big deal tho since it should only be run once on start so performance isn't a major concern.

Reply:
> Using json seems kinda redundant. The data is not very complex, a newline/comma/space separated file format would suffice. Not a big deal tho since it should only be run once on start so performance isn't a major concern.

https://github.com/ferrumc-rs/ferrumc/pull/140#discussion_r1881363216